lib/SelectedObjectsProvider/useObjectSelector.js

```diff
@@ -0,0 +1,49 @@
+import { useContext } from 'react';
+import { SelectedObjectsContext } from './index';
+
+export function useObjectSelector(
+  id,
+  type,
+  idsToSelect,
+  disabledCallback = () => false
+) {
+  const {
+    selected,
+    selectMultiple,
+    deselectMultiple,
+    updateSelected,
+    currentSelectedType,
+    intendedToSelect,
+    setIntendedToSelect
+  } = useContext(SelectedObjectsContext);
+
+  const isSelected = selected.indexOf(id) !== -1;
+  const isDisabled = disabledCallback(currentSelectedType, isSelected);
+  const isHovered = intendedToSelect.indexOf(id) !== -1;
+
+  const getIds = (additionalIds = []) => [
+    ...new Set([id, ...idsToSelect, ...additionalIds])
+  ];
+
+  const handleClick =
+    idsToSelect.length > 1
+      ? () =>
+          isSelected
+            ? deselectMultiple(getIds(), type)
+            : selectMultiple(getIds(), type)
+      : () => updateSelected(idsToSelect[0]);
```

Comment:
should this just be `() => updateSelected(id)`?

Reply:
We wanted to seperate the first argument, `id`, with the ones being selected, `idsToSelect`. This is so we're certain that it will only selected the id's that are in `idsToSelect` and nothing more.

Reply:
Forgive me if im missing something! is individual selection not meant to work for `stories/webapp/hierarchy/HierarchyFolderRow.js` and only for its parent folder? , cos you're not passing over any idsToSelect (just an empty array) so that updateSelected will have nothing to pass up

Reply:
Correct me if I am incorrect @DavidHollins6 but didn't we decide to just use `idsToSelect` rather than the complexity of figuring out "why do we only use the first key in this scenario"?